### ISSUE TITLE
The `agent.conf` validation from Wazuh Manager side may be inaccurate

### ISSUE THREAD

**Issue by jnasselle** [wazuh/wazuh#21742]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|up to Wazuh v4.7.2| verify-agent-conf | Manager | Any | - |

# Description

A community user in [Wazuh Discord channel ](https://discord.com/channels/1049711339578331186/1049711340316541004/1202946236563464222) reports that winagent CIS-CAT configuration could not be saved using Wazuh Dashboard showing the next error

```
Could not update configuration (1908) - Error validating configuration: Unsupported CIS-CAT Binary 'Assessor-CLI.bat'., (1202): Configuration error at 'etc/ossec.conf'.
Error: Could not update configuration (1908) - Error validating configuration: Unsupported CIS-CAT Binary 'Assessor-CLI.bat'., (1202): Configuration error at 'etc/ossec.conf'. 
    at Function.returnErrorInstance (https://192.168.120.64/47202/bundles/plugin/wazuh/wazuh.plugin.js:1:186528)
    at Function._callee2$ (https://192.168.120.64/47202/bundles/plugin/wazuh/wazuh.plugin.js:1:184748)
    at tryCatch (https://192.168.120.64/47202/bundles/plugin/customImportMapDashboards/customImportMapDashboards.plugin.js:13:786910)
    at Generator.invoke [as _invoke] (https://192.168.120.64/47202/bundles/plugin/customImportMapDashboards/customImportMapDashboards.plugin.js:13:790926)
    at Generator.next (https://192.168.120.64/47202/bundles/plugin/customImportMapDashboards/customImportMapDashboards.plugin.js:13:788105)
    at asyncGeneratorStep (https://192.168.120.64/47202/bundles/plugin/wazuh/wazuh.plugin.js:1:178377)
    at _next (https://192.168.120.64/47202/bundles/plugin/wazuh/wazuh.plugin.js:1:178688)
```

## RCA

It was found that Wazuh Dashboard via Wazuh API uses Wazuh Server `verify-agent-conf` tool to validate an agent configuration before saving it and then sharing it using the Centralized configuration mechanism

```xml
<agent_config>
<wodle name="cis-cat">

  <disabled>no</disabled>
  <timeout>1800</timeout>
  <wday>monday</wday>
  <time>04:00</time>
  <interval>2w</interval>
  <scan-on-start>yes</scan-on-start>

  <java_path>\tes-workstationIM1ISTE014\Program Files\Java\jre-1.8\bin</java_path>
  <ciscat_path>C:\cis-cat</ciscat_path>
  <ciscat_binary>Assessor-CLI.bat</ciscat_binary>
  <content type="xccdf" path="benchmarks\your_windows_benchmark_file_xccdf.xml">
     <profile>"Level 2 - Server"</profile>
  </content>
</wodle>
</agent_config>
```

The `verify-agent-conf` tool behavior will change depending on its compilation OS. This causes the next code snippet to validate a valid configuration wrongly

https://github.com/wazuh/wazuh/blob/b28ebb9172d16e417ddc78ba9e73e2fc2a901418/src/config/wmodules-ciscat.c#L183-L197

TL;DR: Wazuh Manager is validating every agent's config as Linux systems and may cause problems with validations that are restricted to compilation time for other OS.

**NOTE:** This issue focuses on the Centralized Configuration issue, not the CIS-CAT module itself.

## Other related issues

- https://github.com/wazuh/wazuh/issues/7308

**Comment by TomasTurina** [comment]
The modules involved will be refactored/deleted, so this issue will not be addressed.
